index.js

```diff
@@ -1,21 +1,21 @@
-// Inspired by (but complete rewrite of) requirejs-metagen
-let _ = require('lodash/fp')
+// Inspired by (but complete rewrite of) requirejs-metagen.
+let _       = require('lodash/fp')
 let Promise = require('bluebird')
-let readDir = Promise.promisify(require('recursive-readdir'))
-let fs = Promise.promisifyAll(require('fs'))
+let readDir = Promise.promisify(require('recursive-readdir')) // Lists all files in a certain directory and all its sub-directories.
+let fs      = Promise.promisifyAll(require('fs'))
 
-// Path Utils
-let filesRelative = (dir, ex) => readDir(dir, ex).map(x => x.slice(dir.length))
+// Define path Utils.
+let relativeFilenames = (dir, exclusions) => readDir(dir, exclusions).map(n => n.slice(dir.length))
 let noExt = file => file.slice(0, _.lastIndexOf('.', file))
-let test = x => y => x.test(y)
+let test = regex => str => regex.test(str)
 
-// Core
-let filter = _.filter(test(/.js|.html|.jsx|.ts|.coffee|.less|.css|.sass|.hbs|.ejs/))
-let metagen = dir => filesRelative(dir.path, dir.exclusions || [dir.output || '__all.js'])
+// Core.
+let filter = _.filter(test(/.js|.html|.jsx|.ts|.coffee|.less|.css|.sass|.hbs|.ejs/)) // Create a default filter.
```

Comment:
This comment is kinda useless as well - if anything we should rename this to `defaultFilter`